### ISSUE TITLE
search: exact dataset title match

### ISSUE THREAD

**Issue by tiborsimko** [cernopendata/opendata.cern.ch#3486]
**Current behaviour**

Consider a user searching for a CMS dataset with a known dataset title such as `/EG/Run2010A-Apr21ReReco-v1/AOD`.
The user copies-pastes the dataset name from a paper into the search box and expects to find the record.

(1) In production, an invalid query s obtained: https://opendata.cern.ch/search?page=1&size=20&q=%2FEG%2FRun2010A-Apr21ReReco-v1%2FAOD

![oops-prod](https://github.com/cernopendata/opendata.cern.ch/assets/517546/6c7292ed-6d5e-4da5-a286-ea27c7669d2b)

(2) We have improved the situation during the previous Invenio upgrade sprint in 2021, see #2930. Using master branch at commit 1b8cd2257, the exact dataset title search works nicely: http://127.0.0.1:5000/search?q=%2FEG%2FRun2010A-Apr21ReReco-v1%2FAOD&l=list&order=asc&p=1&s=10&sort=bestmatch

![oops-master-old](https://github.com/cernopendata/opendata.cern.ch/assets/517546/1023aff5-d332-4f7f-a386-9d1d034a42da)

(3) However, the recent Invenio upgrade (#3450) made things not work again: https://opendata-dev.cern.ch/search?q=%2FEG%2FRun2010A-Apr21ReReco-v1%2FAOD&l=list&order=asc&p=1&s=10&sort=bestmatch

![oops-master-new](https://github.com/cernopendata/opendata.cern.ch/assets/517546/e5d6700a-5771-4ac4-baac-88dfffcb38f4)

**Expected behaviour**

We should find the given dataset when the user copies-pastes the exact dataset title as in 2 above.  See remarks in #2930.

**Comment by psaiz** [comment]
Assuming that we start from the fix of the [AND](https://github.com/cernopendata/opendata.cern.ch/issues/3140),  something like:
```
+++ b/cernopendata/config.py
@@ -215,7 +215,7 @@ RECORDS_REST_ENDPOINTS["recid"]["search_index"] = "records"
 def _query_parser_and(qstr=None):
     """Parser that uses the Q() with AND from search engine dsl."""
     if qstr:
-        return dsl.Q("query_string", query=qstr, default_operator="AND")
+        return dsl.Q("query_string", query=qstr.replace("/", "\\/"), default_operator="AND")
     return dsl.Q()
```

would do the trick. 

Note that if the user escapes the characters, then the page will return no results.

![Screenshot 2023-12-11 at 17 39 45](https://github.com/cernopendata/opendata.cern.ch/assets/12909623/fc56cb09-bf6e-4f10-b11f-39626e028699)
![Screenshot 2023-12-11 at 17 39 36](https://github.com/cernopendata/opendata.cern.ch/assets/12909623/79460a8c-688b-47ca-8a89-e2c4a9dac3a8)

**Comment by psaiz** [comment]
I was considering several options:
1. Escaping the `/`, as mentioned in the previous comment
2. Changing the "query_string" by a "simple_query_string". Discarded this idea, since it would change some of the other semantics
3. Do the standard query, and, if it fails, do one of the previous actions. Discarded this idea as well, since it would depend then on the number of `/` (queries with an even number of those are correct).

So, long story short, the escaping sounds to me as the best option. Incoming PR with that...
